torchmetrics/text/rouge.py

```diff
@@ -0,0 +1,144 @@
+        rouge_keys:
+            A list of rouge types to calculate.
+        decimal_places:
+            The number of digits to round the computed the values to.
+        compute_on_step:
+            Forward only calls ``update()`` and returns None if this is set to False. default: True
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step.
+        process_group:
+            Specify the process group on which synchronization is called. default: None (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When `None`, DDP
+            will be used to perform the allgather.
+
+    Example:
+
+        >>> targets = "Is your name John".split()
+        >>> preds = "My name is John".split()
+        >>> rouge = ROUGEScore()   # doctest: +SKIP
+        >>> from pprint import pprint
+        >>> pprint(rouge(preds, targets))  # doctest: +NORMALIZE_WHITESPACE +SKIP
+        {'rouge1_fmeasure': 0.25,
+         'rouge1_precision': 0.25,
+         'rouge1_recall': 0.25,
+         'rouge2_fmeasure': 0.0,
+         'rouge2_precision': 0.0,
+         'rouge2_recall': 0.0,
+         'rougeL_fmeasure': 0.25,
+         'rougeL_precision': 0.25,
+         'rougeL_recall': 0.25,
+         'rougeLsum_fmeasure': 0.25,
+         'rougeLsum_precision': 0.25,
+         'rougeLsum_recall': 0.25}
+
+    References:
+        [1] ROUGE: A Package for Automatic Evaluation of Summaries by Chin-Yew Lin https://aclanthology.org/W04-1013/
+    """
+
+    def __init__(
+        self,
+        newline_sep: bool = False,
+        use_stemmer: bool = False,
+        rouge_keys: Tuple[str] = ("rouge1", "rouge2", "rougeL", "rougeLsum"),  # type: ignore
+        decimal_places: int = 4,
+        compute_on_step: bool = True,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Optional[Callable] = None,
+    ):
+        super().__init__(
+            compute_on_step=compute_on_step,
+            dist_sync_on_step=dist_sync_on_step,
+            process_group=process_group,
+            dist_sync_fn=dist_sync_fn,
+        )
+
+        if not (_NLTK_AVAILABLE or _ROUGE_SCORE_AVAILABLE):
+            raise ValueError(
+                'ROUGE metric requires that both nltk and rouge-score is installed.'
+                'Either as `pip install torchmetrics[text]` or `pip install nltk rouge-score`'
+            )
+
+        self.newline_sep = newline_sep
+        self.rouge_keys = rouge_keys
+        self.use_stemmer = use_stemmer
+        self.aggregator = BootstrapAggregator()
+        self.scorer = RougeScorer(rouge_keys, use_stemmer=self.use_stemmer)
+        self.decimal_places = decimal_places
+
+    def update(self, preds: List[str], targets: List[str]) -> None:  # type: ignore
```

Comment:
to be consistent with the typing in the functional version
```suggestion
    def update(self, preds: Union[str, List[str]], targets: Union[str, List[str]]) -> None:  # type: ignore
```